packages/tpetra/core/src/Tpetra_BlockCrsMatrix_def.hpp

```diff
@@ -3754,6 +3760,7 @@ class GetLocalDiagCopy {
 
     // The code below works on host, so use a host View.
     auto diagOffsetsHost = Kokkos::create_mirror_view (diagOffsets);
+    // DEEP_COPY REVIEW - DEVICE-TO-HOST
```

Comment:
This method is getLocalDiagCopy. I think that this code can be trivially done on device. For this moment, the code requires the device-to-host deep copy because it uses host method of the diag vector. The code does not seem to be efficient and it is serial. What I mean is that by extracting diagonals on device, this deep copy is not really necessary. Also the code check host_sync only with the debug mode. With a dual view context, why is this users' responsibility ? It might also be necessary to discuss why we need to follow the last touch rule on a gpu system.